File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.py
from sys import *

s = stdin.readline()[:-1]

tmp = []
result = ""
lock = False

def emit(tmp:list) -> str :
    ret = ""
    while tmp:
        ret += tmp.pop()
        
    return ret
    



for c in s :
    if c == "<":
        
        result += emit(tmp)
        lock = True
        result += c
        continue
        
    elif c== ">":
        lock = False
        result += c
        continue
    
    if lock:
        result += c
        
    else:
        if c == " ":
            result += emit(tmp)
            result += " "
        
        else: 
            tmp += c
        
    
            

result += emit(tmp)

print(result)
    
    
    

File: 백준/Silver/17413. 단어 뒤집기 2/단어 뒤집기 2.swift
var s = readLine()!

var stk: [String] = []

var tag: Bool = false

var result: [String] = []

func pop() {
    while !stk.isEmpty {
        result.append(stk.popLast()!)
    }
}

for c in s {
    
    if c == ">" {
        result.append(String(c))
        tag = false
        continue
    }
    
    if tag == true {
        result.append(String(c))
        continue
    }
    
    if c == "<" {
        
        pop()
        
        result.append(String(c))
        tag = true
        continue
    }
     
    if c == " " {
        
        pop()
        
        result.append(String(" "))
        continue
        
    }
    
    stk.append(String(c))
    
}

pop()

print(result.joined())
